consensus/ibft/pos.go

```diff
@@ -146,7 +145,10 @@ func (pos *PoSMechanism) preStateCommitHook(rawParams interface{}) error {
 	}
 
 	// Deploy Staking contract
-	contractState, err := stakingHelper.PredeployStakingSC(nil)
+	contractState, err := stakingHelper.PredeployStakingSC(nil, stakingHelper.PredeployParams{
+		MinValidatorCount: stakingHelper.MinValidatorCount,
+		MaxValidatorCount: stakingHelper.MaxValidatorCount,
+	})
```

Comment:
Why is this not something that' settable from the `ibft switch` params?

cc @Kourin1996

Reply:
Why assume that the default values are ok for the user?

Reply:
I think we can append these values on switching and should load from `engine` in genesis.json. So can give min validator count and max validator count when you run `ibft switch`. And PoSHook can load these values as with contract deployment height and PoS beginning height.

Reply:
fixed a91c866e0f6d742a0cc4af63f07196b4703ec96a

Please take a closer look at this one

Reply:
@zivkovicmilos changed to pointers 90c1e58112a6e4841679d805bfa77c5f51eb820e